<launch>
    <!-- 启动USB摄像头节点 -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
        <!-- 自动检测可用的视频设备 -->
        <param name="video_device" value="$(optenv VIDEO_DEVICE /dev/video0)" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <!-- 修改参数适配Chicony摄像头 -->
        <param name="pixel_format" value="mjpeg" />
        <param name="framerate" value="30" />
        <param name="brightness" value="50" />
        <param name="contrast" value="50" />
        <param name="saturation" value="50" />
        <param name="autofocus" value="true" />
        <param name="focus" value="51" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap"/>
        <!-- 禁用相机标定 -->
        <param name="camera_info_url" value=""/>
    </node>

    <!-- 启动颜色追踪节点 -->
    <node name="color_tracker" pkg="track" type="camera_node.py" output="screen" respawn="true">
        <param name="target_color" value="red" />
        <param name="linear_speed" value="0.2" />
        <param name="angular_speed" value="0.2" />
        <param name="min_area" value="500" />
        <remap from="/camera/image_raw" to="/usb_cam/image_raw"/>
    </node>
</launch>
